File: src/main/java/io/openfuture/openmessanger/repository/AttachmentRepository.java
package io.openfuture.openmessanger.repository;

import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations;
import org.springframework.stereotype.Repository;

import lombok.RequiredArgsConstructor;

@Repository
@RequiredArgsConstructor
public class AttachmentRepository {

    private final NamedParameterJdbcOperations jdbcOperations;

    public void save(final String fileName) {
        final String sql = """
                INSERT INTO attachment(name)
                VALUES (:name)
                """;

        final MapSqlParameterSource parameterSource = new MapSqlParameterSource()
                .addValue("name", fileName);

        jdbcOperations.update(sql, parameterSource);
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/repository/AttachmentRepository.kt
package io.openfuture.openmessanger.repository

import lombok.RequiredArgsConstructor
import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations
import org.springframework.jdbc.support.GeneratedKeyHolder
import org.springframework.stereotype.Repository

@Repository
@RequiredArgsConstructor
class AttachmentRepository(
    private val jdbcOperations: NamedParameterJdbcOperations
) {

    fun save(fileName: String?): Int? {
        val sql = """
                INSERT INTO attachment(name)
                VALUES (:name)
                
                """.trimIndent()
        val parameterSource = MapSqlParameterSource()
            .addValue("name", fileName)

        val keyHolder = GeneratedKeyHolder();

        jdbcOperations.update(sql, parameterSource, keyHolder)
        return keyHolder.keys?.get("id") as Int?
    }

}